src/sentry/web/api.py

```diff
@@ -425,7 +438,7 @@ def _dispatch(self, request, helper, project_id=None, origin=None,
         # This is yanking the auth from the querystring since it's not
         # in the POST body. This means we expect a `sentry_key` and
         # `sentry_version` to be set in querystring
-        auth = self._parse_header(request, helper, project)
+        auth = self._parse_auth(request, helper, project)
```

Comment:
What about overriding `auth_from_request` on the `CspApiHelper` instead? Seems like that's more where it logically belongs.